project/frontend/src/components/SwipeMatchForm.js

```diff
@@ -22,47 +24,114 @@ function SwipeMatchForm(props) {
   const { currLocation, locationName } = props;
 
   const history = useHistory();
+  const { firestore } = useContext(FirestoreContext);
   const authContext = useContext(AuthContext);
-  const signedIn =
-    authContext.currentUser.get && authContext.currentUser.get.isSignedIn();
+
+  const user = authContext.currentUser.get;
+  const signedIn = user && user.isSignedIn();
 
   const [username, setUsername] = useState('');
   const [groupId, setGroupId] = useState('');
+  const [creatorCurrentSession, setCreatorCurrentSession] = useState(null);
+  const [isGroupIdValid, setIsGroupIdValid] = useState(true);
```

Comment:
Should we do this the other way and set this to be false initially and then set to true when it's valid?

Reply:
I can see how that seems more intuitive so maybe I should change some stuff around, but the way I'm currently using `isGroupIdValid` works better if the default is true. `isGroupIdValid` is literally only responsible for highlighting the input text box with a red border when it is flipped to true. So when a user tries to join a session, the id is validated through a backend servlet, and if its invalid, the input box gets highlighted red, then when the user focuses the input box again, it goes back to its normal border color.

If I start with false by default, the input box will be highlighted with a red border from when the popup opens to when the user first focuses it, which I think would be strange.

If `isGroupIdValid` was responsible for actually determining whether the user is able to join a session, then it should definitely be false by default, but I don't actually use it for that.

Reply:
Ah i see, wasn't sure if this was responsible for validating the group ID on the backend in any way. Let's maybe add some documentation above clarifying what this does.